ldapauthenticator/ldapauthenticator.py

```diff
@@ -268,11 +295,22 @@ def authenticate(self, handler, data):
         password = data['password']
         # Get LDAP Connection
         def getConnection(userdn, username, password):
-            server = ldap3.Server(
-                self.server_address,
-                port=self.server_port,
-                use_ssl=self.use_ssl
-            )
+            if self.use_ssl:
+                ca_cert = None
+                client_cert = None
+                client_key = None
+                if self.server_ca_file != '':
+                    ca_cert = self.server_ca_file
+                if self.client_certificate_file != '':
+                    client_cert = self.client_certificate_file
+                if self.client_key_file != '':
+                    client_key = self.client_key_file
```

Comment:
Assuming we go for a merge of this PR, we should also simplify this logic to by letting the new traitlets based config have `allow_none=True` and letting the default value be None.